h5py/_selector.pyx

```diff
@@ -179,6 +179,12 @@ cdef class Selector:
                     a = np.asarray(a)
                 if a.ndim != 1:
                     raise TypeError("Only 1D arrays allowed for fancy indexing")
+                # Fix for issue #1847, not sure this is optimal but it works
+                if np.issubdtype(a.dtype, bool):
```

Comment:
```suggestion
                if a.dtype.kind == 'b':
```

This check seemed to be going wrong on some older numpy versions in CI. I don't know what the best way to write it is, but this suggestion is the same way we check for bool arrays elsewhere in the code: https://github.com/h5py/h5py/blob/3051c76afa73ad0c72d2c7c6724fe3617d2fefd8/h5py/_hl/selections.py#L62